### ISSUE TITLE
Enforcing Kramdown and Rouge breaks PHP syntax highlighting

### ISSUE THREAD

**Issue by penman** [github/pages-gem#205]
Pygments supports a special PHP syntax highlighting option called `startinline`. It instructs the highlighter to begin interpreting code as PHP immediately, rather than waiting for an opening `<?php` tag.

It's used like this:

```liquid
{% highlight php startinline %}
echo "Hello, world!";
{% end %}
```

Rouge also supports this, but the syntax is a little different:

    ```php?start_inline=1
    echo "Hello, world!";
    ```

So far, so good — as long as your markdown parser is Redcarpet.

Kramdown's parser messes up, and will interpret the above code as a fenced code block with no language specified, the first line of which is `php?start_inline=1`.

This means that, unless something changes, after 1 May `startinline` syntax highlighting will disappear from GitHub Pages.

The only way to solve this without removing the Kramdown-enforcement, as I can see it, is

1. Get Kramdown's parser fixed (gettalong/kramdown#234)
2. Get a new version of Kramdown released
3. Get a new version of Jekyll released with the Kramdown dependency. (Bear in mind the last down Jekyll updated the version of Kramdown it depended on was in 2014 — there have been six 1.x released since then.)
4. Upgrade GitHub Pages to the new version of Jekyll

This seems unlikely to all be accomplished by May 1.

So, **the only way to avoid a regression in GitHub Pages is likely to be postponing the date Kramdown enforcement begins**.

**Comment by benbalter** [comment]
> . (Bear in mind the last down Jekyll updated the version of Kramdown it depended on was in 2014 — there have been six 1.x released since then.)

GitHub Pages (and thus Jekyll) is using [Kramdown 1.9.0](https://pages.github.com/versions/). Is that not the latest release?

> Upgrade GitHub Pages to the new version of Jekyll

Which version are you referring to? (We're currently running 3.0.3). Do you have a link to an issue where this bug was fixed?

**Comment by penman** [comment]
Okay, sorry about that, I misread Jekyll's gemspec.

In that case, all that needs to happen is for Kramdown to fix the issue, release a new version and for GitHub Pages to update to the new version of Kramdown.

That seems far more doable. ):

**Comment by svivian** [comment]
+1

**Comment by gettalong** [comment]
This is now fixed in kramdown.

**Comment by parkr** [comment]
kramdown commit: https://github.com/gettalong/kramdown/commit/0f4d1ffade77eda7c0597f4f2535f0ab81f6235c & issue: https://github.com/gettalong/kramdown/issues/234

@gettalong Let us know what your plans are for a new release and we can coordinate on bumping it.

**Comment by gettalong** [comment]
@parkr I will probably release a new version at the end of next week.

**Comment by mmoll** [comment]
#230

**Comment by penman** [comment]
This appears to have been fixed in #230. :+1:
